bin/build-lib.sh

```diff
@@ -0,0 +1,207 @@
+# $1: message
+die () {
+  >&2 echo -e "Error: $1"
+  exit 1
+}
+
+# $1: command
+function run_verbose() {
+  echo "$*"
+  bash -c "$*"
+}
+
+has_item () {
+  for i in $1
+  do
+    if test "$i" = "$2"
+    then
+      echo "$i"
+      break
+    fi
+  done
+}
+
+#------------------------------------------------------------------------------
+# target groups
+#------------------------------------------------------------------------------
+
+test_only () {
+  if test -n "$RUNNING_TESTS"
+  then
+    echo $1
+  fi
+}
+
+bench_only () {
+  if test -n "$RUNNING_BENCHMARKS"
+  then
+    echo $1
+  fi
+}
+
+dev_build () {
+  if test -n "$RUNNING_DEVBUILD"
+  then
+    echo $1
+  fi
+}
+
+# A group consisting of all known individual targets
+all_grp () {
+  { for i in $GROUP_TARGETS
+    do
+      eval "echo \$$i"
```

Comment:
Why can't you echo normally here?

Reply:
We have to echo the values of the variable names.